src/autoscaler/metricscollector/collector/apppoller_test.go

```diff
@@ -211,7 +211,7 @@ var _ = Describe("Apppoller", func() {
 					Expect(appid).To(Equal("test-app-id"))
 					Expect(token).To(Equal("bearer test-access-token"))
 
-					if noaa.ContainerEnvelopesCallCount()%2 == 0 {
+					if noaa.ContainerEnvelopesCallCount() >= 2 && noaa.ContainerEnvelopesCallCount() <= 4 {
```

Comment:
How is this different than the fail and retry?

I would also recommend that we do not rely on logging for testing outcomes. It would be far better to use the fact that we call models.GetInstanceMemoryMetricFromContainerEnvelopes as a means to verify.